### PR TITLE
refactor(GRO-389): inquiry modal form

### PR DESCRIPTION
**Pull request by dzucconi**
So this begins the rebuild of the artwork inquiry modal. This PR handles the actual inquiry for both logged in and logged out users.

When a user is logged in they see the inquiry form, they submit it and it sends.

When a user is logged out they see the inquiry form, along with name/email inputs, they click send and we check to see if there's an account already associated with the email. If so: ask the user to login or reset their password. If not: sign up the user. Once either logged in or signed up, the inquiry sends.

-----

### TODO
- [ ] ReCaptcha impressions
- [ ] Better toast/notifications
- [ ] Improve error messaging

-----

### Logged out

https://user-images.githubusercontent.com/112297/129398927-20c70ffd-de7c-45de-a54a-73e80bb30a11.mov

### Logged in

https://user-images.githubusercontent.com/112297/129399037-63f2c4fb-6d63-48b1-94e0-0205b64e3594.mov

### CODE REVIEW

**Comment by dzucconi** [comment]
OK — it actually appears we no longer allow logged out inquiries — we check to see if the email is associated with the account and ask them to login, or prompt for a sign up otherwise.

File: src/v2/Components/ArtworkInquiry/ArtworkInquiry.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            return <ArtworkInquiryLogin />
+          case Screen.ExistingUser:
+            return <ArtworkInquiryExistingUserQueryRenderer />
+          case Screen.SignUp:
+            return <ArtworkInquirySignUp />
+          case Screen.ResetPassword:
+            return <ArtworkInquiryResetPassword />
+        }
+      })()}
+    </ArtworkInquiryDialog>
```
Comment (damassi) [inline src/v2/Components/ArtworkInquiry/ArtworkInquiry.tsx:48]: Nice and clean pattern here

**Comment by dzucconi** [comment]
Some minor things left here but this should be good to merge (won't be called anywhere yet). I'm going to switch to the questionnaire part and then once that's done give everything a final Q/A + layer of polish.

File: src/v2/Components/ArtworkInquiry/ArtworkInquiry.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      </ArtworkInquiryBackdrop>
+    </ArtworkInquiryProvider>
+  )
+}
+
+interface UseArtworkInquiry {
+  artworkID: string
+}
+
+export const useArtworkInquiry = ({ artworkID }: UseArtworkInquiry) => {
```
Comment (damassi) [inline src/v2/Components/ArtworkInquiry/ArtworkInquiry.tsx:122]: Might be helpful to move this into its own file if used in a few places?

**Review by damassi — APPROVED** [review]
Excellent work 👍

**Pull request merged**